### DISCUSSION TITLE
How to define a heterogeneous array with a specific number of items using JSONSchema (draft/2019-09)?

### DISCUSSION THREAD

**Discussion by busticated** [koxudaxi/datamodel-code-generator#2071] [Q&A]
i'd like to be able to specify an array which can only contain specific types, in exact order, with exact count and have the generated Pydantic models enforce that via validation.  for example:

```js

// valid
[
    "this is a string",
    "this is another string",
    200,
    true
]

// invalid
[
    {},
    "this is a string",
    false,
]

// invalid
[
    {},
    "this is another string",
    200,
    true
]

// invalid
[
    {},
    "this is another string",
    200,
    true,
    "surprise!"
]
```

...which is to say, my array / list should only ever have 4 entries. indices 0 and 1 should be strings. index 2 should be a number. index 3 should be a boolean. anything else should be flagged as invalid. 

is this possible or does it require the as yet unsupported `prefixItems` feature ([issue#1546](https://github.com/koxudaxi/datamodel-code-generator/issues/1546))

**Comment by koxudaxi** [comment]
prefixItems is already supported — use "items": false with minItems/maxItems equal to the prefixItems length to generate tuple[str, str, float, bool] (see PR #2537). Thank you!!
